### ISSUE TITLE
Apparent tunnel disconnection & a reconnection with (visually) missing network parameters 

### ISSUE THREAD

**Issue by X8716e** [AdguardTeam/AdguardForiOS#748]
No reproduction steps currently possible as I noticed it after the fact. Screenshots attached. 
Thumbnail code has currently escaped my mind; please excuse use of full size photos. As is visible, while in ‘full-tunnel with vpn icon mode’ there was an interruption of the tunnel that resulted in a reconnection with both missing network parameters & no visible VPN icon. After a manual disconnect/reconnect, it returned to normal. Thank-you.

![80f36d40-01cb-41b1-aed9-3238e7ec02b9](https://user-images.githubusercontent.com/18148748/38625399-643302e4-3d78-11e8-8770-128e47ea834a.png)
![2bed9da6-56e1-4301-a7c2-39258645c9db](https://user-images.githubusercontent.com/18148748/38625400-65030840-3d78-11e8-8082-d7cb8e670a25.png)
![3b50b3aa-2c6b-482a-863d-6cf2f7536ab6](https://user-images.githubusercontent.com/18148748/38625401-6528cb16-3d78-11e8-94ca-2a4417563a33.jpeg)

**Comment by X8716e** [comment]
It’s now happened - is occurring presently - on my iPad as well. The only influencing factor that may have been the ‘why’ this is occurring was my clearing & resetting of the DNS logs. If it was not this action, then it’s quite a coincidence. I’ve not dis/reconnected this time & the DNS stats are still @ 0 & the logs are still empty. Ad blocking is functioning normally but the DNSCrypt connection has been dropped, reverting back to the iPad’s DNS servers despite what the settings show in the screenshots below.

![784d1ce4-806c-4434-b538-d9e85174e069](https://user-images.githubusercontent.com/18148748/38692906-481e1d4c-3e53-11e8-8f68-fc97b8e8f2d0.png)
![8c77b66f-a76f-440c-b9ed-3654f1f56a59](https://user-images.githubusercontent.com/18148748/38692907-483ba5f6-3e53-11e8-8885-70db925fc0ff.png)
![01982367-6fc6-4701-9832-2ecae3aebd7a](https://user-images.githubusercontent.com/18148748/38692908-485e333c-3e53-11e8-9139-a99b0d79ef7d.png)

**Comment by X8716e** [comment]
logs from phone since install:
[ProTunnel-2018-04-11-223132.ips.beta.zip](https://github.com/AdguardTeam/AdguardForiOS/files/1911548/ProTunnel-2018-04-11-223132.ips.beta.zip)

**Comment by X8716e** [comment]
Any updates on this issue, yet?

**Comment by X8716e** [comment]
3 logs from 04.20 : 
[ProTunnel-2018-04-20-154609.ips.beta.zip](https://github.com/AdguardTeam/AdguardForiOS/files/1934725/ProTunnel-2018-04-20-154609.ips.beta.zip)

**Comment by ameshkov** [comment]
Sorry for the late reply :(

Any change on the latest beta?

**Comment by ameshkov** [comment]
One more question: do you see this error if you use another DNSCrypt server or not use a custom DNSCrypt server at all?

**Comment by X8716e** [comment]
Hello, to answer your latter question - yes, I see it no matter the DNSCrypt server used. It was the very first test that I tried. As to your initial question, I saw it happen with the same frequency in the beta just prior to the current beta. The current beta seems more stable; it’s thus far not shown the same susceptibility. And I’ve been keeping an active watch.

**Comment by ameshkov** [comment]
Thank you! Let's keep an eye on it then, I hope the latest beta is stable enough.

**Comment by X8716e** [comment]
Indeed, let’s. I’ll let you know the first signs of instability if any arise. I’m hopeful, though, that since the thus far none have that none will.

Thank-you.

**Comment by zznosar** [comment]
Got this issue in the latest version on in Full Tunnel Mode in iphone X. When i tested the beta, i was on wifi all the time so i did not encounter the issue in my ipad. Now that I am using the version in my iphone, i lost connectivity everytime i jump from wifi to cellular in Full Tunnel Mode. Split Tunnel seems to be fine.

**Comment by zebrum** [comment]
@zznosar  what about logs? Have you sent it?
